crates/ra_parser/src/grammar/patterns.rs

```diff
@@ -260,11 +261,13 @@ fn pat_list(p: &mut Parser, ket: SyntaxKind) {
 //     let ref mut d = ();
 //     let e @ _ = ();
 //     let ref mut f @ g @ _ = ();
+//     let box i = Box::new(1i32);
 // }
 fn bind_pat(p: &mut Parser, with_at: bool) -> CompletedMarker {
     let m = p.start();
     p.eat(T![ref]);
     p.eat(T![mut]);
+    p.eat(T![box]);
```

Comment:
The `box` must be eated before the rest.

```
let box ref mut a = Box::new(());
```

is valid rust. Proof: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=d2775a785a658a922cdc958e76f8dfa3

Reply:
thanks ❤️